### PR TITLE
Relay Chain

### PR DESCRIPTION
**Pull request by drgora**
This PR supersedes #127.

This PR adds implementations for a relay chain node, and a parachain node. With this PR, the structure of the repository becomes the following:
- mainchain crate: The solo chain `zkv-node`, which should be compatible with the ZKV testnet
- zkv-relay crate: The relay chain `zkv-relay`, which integrates the solo chain node with support for parachains
- paratest crate: The parachain node `paratest-node`, provided as parachain example *for development only*

Instructions to build the solo chain node have been updated in `README.md`. Instructions to build and run the new components are in `RELAY.md`.

The runtime is shared for both chain and the `relay` feature is used to enable the relay chain specific features.

### CODE REVIEW

File: .github/scripts/mount_disk.sh [hunk 1]
```diff
@@ -19,27 +19,22 @@
 # cached docker images.
 #
 echo "=============================================================================="
```
Comment (drgora) [inline .github/scripts/mount_disk.sh:21]: Forgot to remove this entire file, consider it done already.

**Review by DanieleDiBenedetto — APPROVED** [review]
LGTM !!!!!!!!!!!!!

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -842,6 +897,95 @@ construct_runtime!(
...
+
+        // Utility modules.
+        Utility: pallet_utility = 30,
+        Multisig: pallet_multisig = 31,
+        Proxy: pallet_proxy = 32,
+
+
+        // Pallets that we know are to remove in a future. Start indices at 50 to leave room.
+        Sudo: pallet_sudo = 50,
+        ImOnline: pallet_im_online = 51,
```
Comment (la10736) [inline runtime/src/lib.rs:947]: Maybe we can remove it?

File: runtime/src/paratest_genesis
Comment (la10736) [inline runtime/src/paratest_genesis:1]: We should not commit it. We should review a bit the migration code to get the file in a `staging` folder that we can add to `.gitignore`.

The review the `RELAY.md`

File: runtime/src/paratest_wasm
Comment (la10736) [inline runtime/src/paratest_wasm:1]: See above comment

File: native/Cargo.toml [hunk 3]
```diff
@@ -34,4 +34,5 @@ std = [
     "dep:zksync-era-verifier-deserialize",
     "dep:risc0-verifier",
     "dep:ultraplonk_verifier",
+    "hp-verifiers/std",
```
[the remark this answers is not in the document]
Comment (drgora) [reply native/Cargo.toml:37]: Good catch, removed the duplicate

File: runtime/src/paratest_genesis
Comment (drgora) [reply runtime/src/paratest_genesis:1]: Good point, done

File: pallets/verifiers/Cargo.toml [hunk 4]
```diff
@@ -38,7 +38,7 @@ default = ["std"]
 std = [
     "codec/std",
     "frame-support/std",
-    "frame-system/std",
+    "frame-system/default",
```
[the remark this answers is not in the document]
Comment (drgora) [reply pallets/verifiers/Cargo.toml:41]: It looks like we don't, moved back to `std`

File: primitives/hp-verifiers/Cargo.toml [hunk 5]
```diff
@@ -24,4 +24,5 @@ std = [
     "sp-core/std",
     "sp-std/std",
     "sp-io/std",
+    "sp-io/default",
```
[the remark this answers is not in the document]
Comment (drgora) [reply primitives/hp-verifiers/Cargo.toml:27]: It looks like we don't, removed it

File: runtime/src/lib.rs (hunk 2 above)
Comment (drgora) [reply runtime/src/lib.rs]: I believe we should really consider this change carefully and see if it aligns well with our use case.
Anyways, for now I'm ok with keeping this aligned with Polkadot, and I removed it.

File: relay-node/service/chain-specs/zkverify_testnet.json
[the remark this answers is not in the document]
Comment (drgora) [reply relay-node/service/chain-specs/zkverify_testnet.json:1]: Yeah I agree, removed

File: .gitignore [hunk 6]
```diff
@@ -11,6 +11,7 @@ zombienet-tests/yarn.lock
 zombienet-tests/bin
 lcov.info
 benchmarking_errors.txt
+staging/
```
Comment (la10736) [inline .gitignore]: Just a question if you add this here it should not possible to remove it but, the `.gitignore` in `staging` folder that ignore all files instead `.gitignore` should already do the job.

File: .gitignore [hunk 7]
```diff
@@ -21,3 +22,5 @@ coverage_summary.txt
 
 # Ignore external libraries
 deps/
+runtime/src/paratest_genesis
+runtime/src/paratest_wasm
```
Comment (la10736) [inline .gitignore]: If I saw well you've removed the migration parachain logic...Maybe we can remove this too.

**Review by la10736 — APPROVED** [review]
LGTM

**Pull request merged**